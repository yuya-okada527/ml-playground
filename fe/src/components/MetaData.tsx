import React from "react";
import Head from "next/head";
import theme from "../utils/theme";

type MetaDataProps = {
  title: string;
};

/**
 * headタグを記述する
 *
 * @param title タイトル
 */
const MetaData: React.FC<MetaDataProps> = ({ title }) => (
  <Head>
    <title>{title}</title>
    <meta charSet="utf-8" />
    <meta name="viewport" content="initial-scale=1.0, width=device-width" />
    <meta name="theme-color" content={theme.palette.primary.main} />
    <link
      rel="stylesheet"
      href="https://fonts.googleapis.com/css?family=Roboto:300,400,500,700&display=swap"
    />
    <link rel="preconnect" href="https://fonts.gstatic.com" />
    <link
      href="https://fonts.googleapis.com/css2?family=Noto+Sans+JP&display=swap"
      rel="stylesheet"
    />
  </Head>
);

export default MetaData;
