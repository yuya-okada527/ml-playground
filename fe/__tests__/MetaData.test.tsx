import React from "react";
import { shallow } from "enzyme";
import MetaData from "../src/components/MetaData";

describe("MetaData", () => {
  it("Snapshot Test", () => {
    const metaData = shallow(<MetaData title={"title"} />);
    expect(metaData).toMatchSnapshot();
  });
});
