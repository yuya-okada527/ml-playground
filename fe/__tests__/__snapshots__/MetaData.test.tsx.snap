// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MetaData Snapshot Test 1`] = `
<Head>
  <title>
    title
  </title>
  <meta
    charSet="utf-8"
  />
  <meta
    content="initial-scale=1.0, width=device-width"
    name="viewport"
  />
  <meta
    content="#3f51b5"
    name="theme-color"
  />
  <link
    href="https://fonts.googleapis.com/css?family=Roboto:300,400,500,700&display=swap"
    rel="stylesheet"
  />
  <link
    href="https://fonts.gstatic.com"
    rel="preconnect"
  />
  <link
    href="https://fonts.googleapis.com/css2?family=Noto+Sans+JP&display=swap"
    rel="stylesheet"
  />
</Head>
`;
